dont-wreck-my-house/src/main/java/learn/lodging/domain/ReservationService.java

```diff
@@ -0,0 +1,199 @@
+            return result;
+        }
+        reservation.setGuest(guestRepository.findById(reservation.getGuestId()));
+        reservation.setHost(hostRepository.findById(reservation.getHostId()));
+        LocalDate start = reservation.getStartDate();
+        LocalDate end = reservation.getEndDate();
+        reservation.setTotal(calculateTotal(start, end, reservation.getHostId()));
+
+        result.setPayload(reservationRepository.add(reservation));
+        return result;
+    }
+
+    public boolean updateHost(Host host) throws DataException {
+        return reservationRepository.updateHost(host);
+    }
+
+    public boolean updateGuest(Guest guest) throws DataException {
+        return reservationRepository.updateGuest(guest);
+    }
+
+    public boolean deleteGuest(Guest guest) throws DataException {
+        return reservationRepository.deleteGuest(guest);
+    }
+
+    public boolean deleteHost(Host host) throws DataException {
+        return reservationRepository.deleteHost(host);
+    }
+
+    public Result<Reservation> update(Reservation reservation) throws DataException {
+        Result<Reservation> result = validate(reservation);
+        if (!result.isSuccess()){
+            return result;
+        }
+        List<Reservation> reservations = reservationRepository.findByHostId(reservation.getHostId());
+        Reservation existing = null;
+        for (Reservation r: reservations){
+            if (r.getId() == reservation.getId()){
+                existing = r;
+            }
+        }
+        if (existing == null){
+            result.addErrorMessage("Reservation not found.");
+            return result;
+        }
+        LocalDate start = reservation.getStartDate();
+        LocalDate end = reservation.getEndDate();
+        reservation.setTotal(calculateTotal(start, end, reservation.getHostId()));
+
+        boolean success = reservationRepository.update(reservation);
+        if (!success){
+            result.addErrorMessage("Reservation not found.");
+        }
+        return result;
+    }
+
+    public Result<Reservation> delete(Reservation reservation) throws DataException {
+        Result<Reservation> result = validate(reservation);
+        if (!result.isSuccess()){
+            return result;
+        }
+        if (reservation == null){
+            result.addErrorMessage("Reservation not found.");
+        }
+
+        boolean success = reservationRepository.delete(reservation);
+        if (!success){
+            result.addErrorMessage("Reservation not found.");
+        }
+        return result;
+    }
+
+    public BigDecimal calculateTotal (LocalDate start, LocalDate end, String hostId){
+        BigDecimal standard = hostRepository.findById(hostId).getStandardRate();
+        BigDecimal weekend = hostRepository.findById(hostId).getWeekendRate();
+        int numStandard = 0;
+        int numWeekend = 0;
+
+        for (;start.compareTo(end) <= 0; start = start.plusDays(1)){
```

Comment:
I think this is slightly off. I was able to make a reservation of 2/10/2022-2/10/2022 which cost me non-zero total. I think this is charging for one extra day.